prometheus/operators/aws-environment.yml

```diff
@@ -0,0 +1,220 @@
+---
+
+# Deploy to the Support Network in AWS
+- type: replace
+  path: /instance_groups/name=alertmanager/networks/name=default/name?
+  value: Support
+
+- type: replace
+  path: /instance_groups/name=prometheus/networks/name=default/name?
+  value: Support
+
+- type: replace
+  path: /instance_groups/name=database/networks/name=default/name?
+  value: Support
+
+- type: replace
+  path: /instance_groups/name=grafana/networks/name=default/name?
+  value: Support
+
+- type: replace
+  path: /instance_groups/name=nginx/networks/name=default/name?
+  value: Support
+
+- type: replace
+  path: /instance_groups/name=firehose/networks/name=default/name?
+  value: Support
+
+# Update storage volume sizes for persistent services
+- type: replace
+  path: /instance_groups/name=alertmanager/persistent_disk_type?
+  value: 10GB
+
+- type: replace
+  path: /instance_groups/name=prometheus/persistent_disk_type?
+  value: 10GB
+
+- type: replace
+  path: /instance_groups/name=database/persistent_disk_type?
+  value: 10GB
+
+- type: replace
+  path: /instance_groups/name=grafana/persistent_disk_type?
+  value: 10GB
+
+# add monitoring vm_extension to instances (for security groups)
+- type: replace
+  path: /instance_groups/name=alertmanager/vm_extensions?
```

Comment:
As written this will replace `vm_extensions` with the list specified, which might be what you intended.

If instead you wanted to append to the list, creating it if it exists, you can do:

```yaml
- type: replace
  path: /instance_groups/name=alertmanager/vm_extensions?/-
  value: monitoring
```

Reply:
i didn't consciously think about whether replacing or appending was more appropriate.  it worked with replacing, but am happy to append if that's a way we're doing it else where.

Reply:
There are pros and cons. The pro is that is means we'll at least fail fast next time the underlying deployment adds a new `vm_extension` that we haven't declared in our `cloud-config`.

The con is that multiple `vm_extensions` interact strangely. They are applied in order, and shallow merged. Meaning that the last that defines security group will override the others.

That said, I think the pro outweighs the con.